### ISSUE TITLE
Mission editor - ResearchTreeEditor should show mission(s) to unlock research

### ISSUE THREAD

**Issue by SenorRagequit** [exorstudios/riftbreaker-tools#117]
**Is your feature request related to a problem? Please describe.**
Currently, some research objects depend on missions to get unlocked, like this energy walls for example:
![grafik](https://user-images.githubusercontent.com/50502395/172655865-487cd48b-0fb9-4665-9462-1b200a67ace6.png)
But nowhere is written what that mission actually is. 
So without any pre-experience, you now have to manually search through every mission file to find where this unlocks.

**Describe the solution you'd like**
In the research requirements, list the path to the mission file(s) which unlocks the research. 
Similar to how it shows building requirements:
![grafik](https://user-images.githubusercontent.com/50502395/172656306-6857e5ec-5f19-47eb-98c0-6e16a97679c0.png)

**Describe alternatives you've considered**
Read each and every of the mission files

**Comment by lukaasm** [comment]
yeah, hard thing to do since node is unlocked by random node in random logic file or random line in random lua script file :(
Without parsing lua/logic files I have no hard-link to asset that I can use
